### PR TITLE
Fix integer conversion warnings in src/core/support

### PR DESCRIPTION
**Pull request by murgatroid99**
Changed a couple of types, mostly added type casts.

### CODE REVIEW

File: src/core/support/cpu_linux.c [hunk 1]
```diff
@@ -48,10 +48,10 @@
 #include <grpc/support/log.h>
 #include <grpc/support/sync.h>
 
-static int ncpus = 0;
+static unsigned ncpus = 0;
 
 static void init_num_cpus() {
-  ncpus = sysconf(_SC_NPROCESSORS_ONLN);
+  ncpus = (unsigned)sysconf(_SC_NPROCESSORS_ONLN);
```
Comment (murgatroid99) [inline src/core/support/cpu_linux.c]: Unfortunately, I think this is wrong. `sysconf` apparently returns `-1` when a value cannot be determined.

File: src/core/support/murmur_hash.c [hunk 2]
```diff
@@ -57,11 +57,11 @@ gpr_uint32 gpr_murmur_hash3(const void *key, size_t len, gpr_uint32 seed) {
...
   const gpr_uint32 c2 = 0x1b873593;
 
-  const gpr_uint32 *blocks = (const uint32_t *)(data + nblocks * 4);
-  const uint8_t *tail = (const uint8_t *)(data + nblocks * 4);
+  const gpr_uint32 *blocks = ((const gpr_uint32 *)key) + nblocks;
+  const gpr_uint8 *tail = (const gpr_uint8 *)(data + nblocks * 4);
 
   /* body */
-  for (i = -nblocks; i; i++) {
+  for (i = -(int)nblocks; i; i++) {
```
Comment (ctiller) [inline src/core/support/murmur_hash.c]: Change type of i to size_t, remove the cast.

File: src/core/support/murmur_hash.c [hunk 3]
```diff
@@ -78,9 +78,9 @@ gpr_uint32 gpr_murmur_hash3(const void *key, size_t len, gpr_uint32 seed) {
   /* tail */
   switch (len & 3) {
     case 3:
-      k1 ^= tail[2] << 16;
+      k1 ^= (gpr_uint32)(tail[2] << 16);
```
Comment (ctiller) [inline src/core/support/murmur_hash.c]: ```((gpr_uint32)(tail[2])) << 16``` seems like it'd be more robust against future changes

File: src/core/support/slice.c [hunk 4]
```diff
@@ -311,7 +313,7 @@ gpr_slice gpr_slice_split_head(gpr_slice *source, size_t split) {
 }
 
 int gpr_slice_cmp(gpr_slice a, gpr_slice b) {
-  int d = GPR_SLICE_LENGTH(a) - GPR_SLICE_LENGTH(b);
+  int d = (int)(GPR_SLICE_LENGTH(a) - GPR_SLICE_LENGTH(b));
```
Comment (ctiller) [inline src/core/support/slice.c]: ssize_t?

File: src/core/support/murmur_hash.c (hunk 2 above)
Comment (murgatroid99) [inline src/core/support/murmur_hash.c]: `i` is always `<=0`.

Comment (ctiller) [inline src/core/support/murmur_hash.c]: oh... that's awful(ly subtle)

File: src/core/support/slice.c (hunk 4 above)
Comment (murgatroid99) [inline src/core/support/slice.c]: done, but I had to include `sys/types.h`.

File: src/core/support/murmur_hash.c (hunk 3 above)
Comment (murgatroid99) [inline src/core/support/murmur_hash.c]: done

File: src/core/support/murmur_hash.c (hunk 2 above)
Comment (murgatroid99) [inline src/core/support/murmur_hash.c]: It's not even necessary. You could just shift everything left by `nblocks` and the negatives. But poking at the functionality of something like this was out of scope of this PR.

File: src/core/support/slice.c (hunk 4 above)
Comment (murgatroid99) [inline src/core/support/slice.c]: Actually, `ssize_t` is a Posix thing. I'm going to revert this.

**Pull request merged**